### ISSUE TITLE
[Question] Could you please provide more details about the publicly available pre-trained models?

### ISSUE THREAD

**Issue by rishiyama** [CaptainDario/DaKanji-Single-Kanji-CNN#2]
Hello, I am using your publicly available model in my research activities.

Could you kindly provide details about the training environment of the latest publicly released model? https://github.com/CaptainDario/DaKanji-Single-Kanji-Recognition/releases/tag/v1.2

I noticed that the pre-trained model uses the ETL Character Database and the KanjiVG dataset. Could you please provide further details about this?

Specifically, I am interested in how KanjiVG was used in this model. For example:

1. How was it provided as input?
2. Were any preprocessing steps applied to the dataset?

Additionally, could you please share the recognition performance (accuracy) of the pre-trained model?

Thank you for releasing this model that handles Japanese characters. I truly appreciate your work.

**Comment by dariyooo** [comment]
Hello,

thank you for your interest in this project and your kind words.

> Could you kindly provide details about the training environment of the latest publicly released model?

The notebook used to train that model can be found [here](https://github.com/CaptainDario/DaKanji-Single-Kanji-Recognition/blob/master/single_kanji_cnn_training.ipynb)

> Specifically, I am interested in how KanjiVG was used in this model. For example:

The KanjiVG dataset contains svg (vector graphics) data. This data has been read and the individual curves have been modified randomly. The results of those modifications were then renderer onto images and fed to the network.

> Additionally, could you please share the recognition performance (accuracy) of the pre-trained model?

I do not have a benchmark value other than what is stored in the [training notebook](https://github.com/CaptainDario/DaKanji-Single-Kanji-Recognition/blob/master/single_kanji_cnn_training.ipynb). But I remember that on the ETLCDB it had accuracy > 0.99.

Hope this helps

**Comment by rishiyama** [comment]
Thank you for your reply!


>The KanjiVG dataset contains svg (vector graphics) data. This data has been read and the individual curves have been modified randomly. 

Regarding the processing of KanjiVG:
1. For a kanji with N strokes, does it read all N `<path>` data, then randomly select M(≦N) elements and randomly perturb their coordinates? 
2. Is there no operation that deletes any strokes? 
3. Are only the original KanjiVG fonts used?

Are the above operations implemented by these functions in the following file?
[DataGenerator.py](https://github.com/CaptainDario/DaKanji-Single-Kanji-Recognition/blob/9d5c1a433c6edae03be8d52f481cf11119839714/DataGenerator.py)

If you remember, I would be grateful if you could let me know.

Thank you for the information.


p.s.

Since the training code does not include any processing related to KanjiVG, I would appreciate it if you could add it to the repository's code if possible.

For example, I believe this commit includes the processing for reading SVGs:
https://github.com/CaptainDario/DaKanji-Single-Kanji-Recognition/commit/597625df6c581353b288499b1d56293d4423a980

**Comment by dariyooo** [comment]
I'm sorry for not getting back to you sooner.

> 1. For a kanji with N strokes, does it read all N <path> data, then randomly select M(≦N) elements and randomly perturb their coordinates?

Yes that is correct.

> 2. Is there no operation that deletes any strokes?

I am not sure about it, but I think no strokes were deleted
 
> 3. Are only the original KanjiVG fonts used?

For this kind of generation only kanjiVG has been used

> Are the above operations implemented by these functions in the following file?

No, the mentioned file is another layer of data augmentation.
In addition to the KanjiVG generation method, I rendered a set of fonts (without individual stroke manipulation) to images. The augmentation methods in that file are applied to those images.

> Since the training code does not include any processing related to KanjiVG, I would appreciate it if you could add it to the repository's code if possible.

I can look if I can still find that old code but I guess I lost it somewhere.

Hope this helps.

**Comment by rishiyama** [comment]
Thanks so much for your response.  I really appreciate the explanation. I wish you the best with your projects!

**Comment by dariyooo** [comment]
Thank you!
